TowersOfHanoi/TowersOfHanoi.cs

```diff
@@ -1,12 +1,270 @@
+
+            Console.Clear();
+            DrawGameBoard();
+
+            while(!won)
+            {
+                String towerOrigin = "";
+                String towerDestination = "";
+
+                try
+                {
+                    towerOrigin = GetUserInput();
+                    // block = PickBlock(towerOrigin);  Feel this is less effective and creates more code to manage
+                    towerDestination = GetUserInput();
+                    MoveBlock(/* block,  */towerOrigin, towerDestination);
+                    Console.Clear();
+                    DrawGameBoard();
+                    won = CheckForWin();
+                }
+                catch
+                {
+                    Console.WriteLine("Invalid move.");
+                }
+            }
+
+            Console.WriteLine("You win!");
+        }
+
+        // Builds the gameboard. Asks the user how may blocks they would like to play with and builds the block
+        // objects in the first tower based on user input. There are a minimum of 4 blocks created.
+        public int CreateGameBoard()
+        {
+            Console.WriteLine("Please enter the number of blocks you want to play with if more than 4: ");
+            int numberOfBlocks = Convert.ToInt32(Console.ReadLine());
+
+            // Creates block objects based on user input
+            if (numberOfBlocks > 4)
+            {
+                gameboard.Add("A", new Tower(numberOfBlocks));
+                gameboard.Add("B", new Tower());
+                gameboard.Add("C", new Tower());
+
+                Console.WriteLine($"You will play with {numberOfBlocks} blocks.");
+                Thread.Sleep(2000);
+
+                return numberOfBlocks;
+            }
+            // Builds the minimum number of block objects for the game
+            else
+            {
+                Console.WriteLine("Game will with be built with the minimum 4 blocks.");
+                gameboard.Add("A", new Tower(4));
+                gameboard.Add("B", new Tower());
+                gameboard.Add("C", new Tower());
+
+                Thread.Sleep(1500);
+
+                return 4;
+            }
+        }
+
+        // Draws the game board
+        public void DrawGameBoard()
+        {
+            String formattedString = "";
+
+            foreach (String key in gameboard.Keys)
+            {
+                formattedString += key + ": " + gameboard[key] + "\n";
+            }
+            Console.WriteLine(formattedString);
+        }
+
+        // Asks the user to select a tower to or from
+        public String GetUserInput()
+        {
+            String towerSelect;
+            do
+            {
+                // towerMove is different depending on whether this is the first or second call of GetUserInput in the 
+                // do-while loop this is called from.
```

Comment:
I see the towerMove being set to "to" but where does it flip back to "from"? did i miss it?

Reply:
never mind i see it now